module/Finna/src/Finna/View/Helper/Root/RecordDataFormatterFactory.php

```diff
@@ -999,29 +999,43 @@ protected function getDefaultCoreFields()
 
         // Add arcrole-relations as multiple fields with role as field header
         $getRelations = function ($data, $options) use (&$pos) {
-            $final = [];
-            foreach ($data as $type => $values) {
-                $label = null;
-                if (isset($values['role'])) {
-                    $label = $values['role'];
-                    $label = "CreatorRoles::$label";
+            // Group relations by role
+            $relationsByRole = $relationsWithoutRole = [];
+            foreach ($data as &$relation) {
+                if ($role = ($relation['role'] ?? null)) {
```

Comment:
Tätä voisi yksinkertaistaa niin, että `$role = ($relation['role'] ?? '0') ?: '0';` (lienee syytä varautua myös siihen, että rooli on määritelty, mutta tyhjä), ja kaikki samaan taulukkoon. Sitten voidaan koota lopullinen taulukko yhdellä kertaa.

Reply:
Kyllä, yksinkertaisempi näin.